lib/saluki-env/src/autodiscovery/mod.rs

```diff
@@ -96,6 +99,49 @@ pub struct Config {
     pub logs_excluded: bool,
 }
 
+impl Config {
+    /// Get the ID for an instance
+    pub fn id(&self, digest: &u64, instance: &HashMap<MetaString, serde_yaml::Value>) -> String {
```

Comment:
You don't need to pass primitives, like integers, by reference unless you need to actually provide a mutable reference so the callee is able to mutate it.

Otherwise, it's not any cheaper: `u64` is 8 bytes, and on 64-bit platforms, so is the pointer backing the reference to it.